File: Baekjoon/Class3/14.11659 구간 합 구하기 4.py
import sys
input = sys.stdin.readline

n, m = map(int, input().split())
out = []

nums = list(map(int, input().split()))
sums = [0] * (n + 1)
for i in range(1, n + 1):
    sums[i] = sums[i - 1] + nums[i - 1]

for _ in range(m):
    i, j = map(int, input().split())
    out.append(str(sums[j] - sums[i - 1]))
print("\n".join(out))

File: Baekjoon/Class3/14.11659 구간 합 구하기 4.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim().split("\n");

const [n, m] = input[0].split(" ").map(Number);
const out = [];

const nums = input[1].split(" ").map(Number);
const sums = new Array(n + 1).fill(0);
for(let i = 1; i <= n; i++)
    sums[i] = sums[i - 1] + nums[i - 1];

for(let k = 2; k < m + 2; k++) {
    const [i, j] = input[k].split(" ").map(Number);
    out.push(sums[j] - sums[i - 1]);
}
console.log(out.join("\n"));